### ISSUE TITLE
fit report problem for small numbers

### ISSUE THREAD

**Issue by mbakker7** [lmfit/lmfit-py#458]
In the code below, the optimal value of parameter `a` is `9.999967403714958e-05`, but it shows up in fit_report as `9.99996740 +/- 7.75e-11`.

```
from lmfit import Parameters, minimize
import numpy as np
p = Parameters()
p.add('a', value=1e-3)
p.add('b', value=10)
ydata = 50 + 1e-4 * np.arange(1, 4) * np.array([1.000001, 0.999999, 0.999998])

def func(p):
    vals = p.valuesdict()  # store dictionary in variable vals
    a = vals['a']
    b = vals['b']
    rv = a * np.arange(1, 4) + b
    return rv - ydata

result = minimize(func, p)
print(fit_report(result))
print('optimal params:', result.params)
```
The code above produces the following output:
```
[[Fit Statistics]]
    # fitting method   = leastsq
    # function evals   = 9
    # data points      = 3
    # variables        = 2
    chi-square         = 0.00000
    reduced chi-square = 0.00000
    Akaike info crit   = -136.90607
    Bayesian info crit = -138.70885
[[Variables]]
    a:   9.99996740 +/- 7.75e-11 (0.00%) (init= 0.001)
    b:   50         +/- 1.67e-10 (0.00%) (init= 10)
[[Correlations]] (unreported correlations are <  0.100)
    C(a, b)                      = -0.926
optimal params: Parameters([('a', <Parameter 'a', value=9.999967403714958e-05 +/- 7.75e-11, bounds=[-inf:inf]>), ('b', <Parameter 'b', value=50.00000000036331 +/- 1.67e-10, bounds=[-inf:inf]>)])
```

**Comment by mbakker7** [comment]
Sorry, I thought this would be obvious in the code. Indeed took me a little time to get a stand-alone example together. I hope it is easy to fix. I use version 0.9.8. Thanks for your help.

**Comment by reneeotten** [comment]
thanks, that indeed reproduces it! I tried something this morning quickly, but didn't have enough decimal points to notice the issue... It's caused by the ```fit_report``` function, we seem to use a fixed-with to print the table. If you change the number of decimal points, the ```e-05``` does show up. I'll take a look and try to fix it later today!

**Comment by newville** [comment]
@mbakker7 @reneeotten  Yikes - that definitely needs a quick fix.   I see this:


```
from lmfit.printfuncs import gformat
>>> print(gformat(9.998e-5))
9.9980e-05
>>> print(gformat(9.9981234e-5))
 9.9981e-05
>>> print(gformat(9.9991234e-5))
 9.9991e-05
>>> print(gformat(9.9999234e-5))
 9.9999e-05
>>> print(gformat(9.9999834e-5))
 9.9999834e
```
but I'm not 100% sure I see how that happens... I'll investigate, hoping to fix soon.

**Comment by newville** [comment]
@mbakker7 @reneeotten  See #459  With that change, I cannot produce any errors.

I'm inclined to merge and tag this (and other recent changes) as version 0.9.9.

**Comment by reneeotten** [comment]
it does mess up the layout a little bit when there are both exponents and floating points...
```
[[Variables]]
    a:   9.99997e-05 +/- 7.75e-11 (0.00%) (init= 0.001)
    b:   50         +/- 1.67e-10 (0.00%) (init= 10)
```
that's not a major issue, but still... if you're in a hurry you can tag now.

I do think though the ```fit_report``` and ```gformat``` functions can be simplified. I looked a little bit, but have not finished... If you're willing to wait until after the weekend then I'll give it a go. Either way, I'll probably take a look anyway and a possible PR can always be merged for a later version.

As a side note, before tagging a new version the documentation should be updated to reflect the ```asteval``` requirement. I have this already in a PR together with some other changes in the documentation. That's not completely ready right now, but I can finish it up in the weekend.

**Comment by newville** [comment]
@reneeotten you're right -- that should be cleaned up so that length and alignment are better preserved.  `gformat` can almost certainly be improved even more. 

I agree that the new requirement for external `asteval` needs documentation, and may cause some pain for people mixing installs with `setup.py install` and `pip`.   I'd like to get 0.9.9 released next week (even early next week), since the fixes since 0.9.8 came fast but are pretty important.

**Comment by reneeotten** [comment]
@newville there are some more issues with the ```gformat``` function in the sense that it adds ```0``` to get the same output length for all variables. However, that's not correct as can be seen in the example below:

```
[[Variables]]
    amp:     5.05591000 +/- 0.038810 (0.77%) (init= 10)
    decay:   0.02548840 +/- 0.000447 (1.76%) (init= 0.1)
    shift:  -0.08528920 +/- 0.009891 (11.60%) (init= 0)
    omega:   1.99728000 +/- 0.003186 (0.16%) (init= 3)
```

the parameters from the fit are:
```
Parameters([('amp', <Parameter 'amp', value=5.055907707955667 +/- 0.0388, bounds=[0:inf]>), 
('decay', <Parameter 'decay', value=0.025488406603568586 +/- 0.000447, bounds=[-inf:inf]>), 
('shift', <Parameter 'shift', value=-0.08528918494699966 +/- 0.00989, bounds=[-1.5707963267948966:1.5707963267948966]>), 
('omega', <Parameter 'omega', value=1.9972843877188868 +/- 0.00319, bounds=[-inf:inf]>)])
```

so extending, for example, ```amp``` to give the same output length should become ```5.05590771``` and not ```5.05591000```. 

I think it should be possible to get this correct by using Python string formatting, but I haven't figured it out yet completely... to be continued.

**Comment by newville** [comment]
@reneeotten I just pushed a new attempt that fixes some problems, and I think ought to fix this too.

**Comment by reneeotten** [comment]
fixed by #459

**Comment by newville** [comment]
@reneeotten thanks, closing this issue
